<a href="https://colab.research.google.com/github/LuppeCh/F1EDA/blob/correcci%C3%B3nEDA/Exploraci%C3%B3n_de_Datos_Trabajo_Final.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
install.packages("dplyr")
install.packages("ggplot2")

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)

Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



In [2]:
library(dplyr)
library(ggplot2)


Attaching package: ‘dplyr’


The following objects are masked from ‘package:stats’:

    filter, lag


The following objects are masked from ‘package:base’:

    intersect, setdiff, setequal, union




In [3]:
wil <- read.csv("williams.csv")

In [4]:
head(wil)

carId  trackId    trackLength lapIndex lapNum lapFlag binIndex validBin
1 Alpine Interlagos 4294        0        0      0       0        1       
2 Alpine Interlagos 4294        0        0      0       1        1       
3 Alpine Interlagos 4294        0        0      0       2        1       
4 Alpine Interlagos 4294        0        0      0       3        1       
5 Alpine Interlagos 4294        0        0      0       4        1       
6 Alpine Interlagos 4294        0        0      0       5        1       
  lap_number lap_distance ⋯ gear_box_damage engine_damage mguh_damage es_damage
1 0          0.513507     ⋯ 0               0             0           0        
2 0          1.449180     ⋯ 0               0             0           0        
3 0          2.384850     ⋯ 0               0             0           0        
4 0          3.320510     ⋯ 0               0             0           0        
5 0          4.256180     ⋯ 0               0             0           0        
6 0          5.194280     ⋯ 0               0             0           0        
  ce_damage ice_damage mguk_damage tc_damage engine_blown engine_seized
1 0         0          0           0         0            0            
2 0         0          0           0         0            0            
3 0         0          0           0         0            0            
4 0         0          0           0         0            0            
5 0         0          0           0         0            0            
6 0         0          0           0         0            0

In [5]:
wil$carId <- "WIL"

In [6]:
tail(wil)

carId trackId    trackLength lapIndex lapNum lapFlag binIndex validBin
25759 WIL   Interlagos 4294        5        5      0       4288     0       
25760 WIL   Interlagos 4294        5        5      0       4289     0       
25761 WIL   Interlagos 4294        5        5      0       4290     0       
25762 WIL   Interlagos 4294        5        5      0       4291     0       
25763 WIL   Interlagos 4294        5        5      0       4292     0       
25764 WIL   Interlagos 4294        5        5      0       4293     0       
      lap_number lap_distance ⋯ gear_box_damage engine_damage mguh_damage
25759 -1         -1           ⋯ -1              -1            -1         
25760 -1         -1           ⋯ -1              -1            -1         
25761 -1         -1           ⋯ -1              -1            -1         
25762 -1         -1           ⋯ -1              -1            -1         
25763 -1         -1           ⋯ -1              -1            -1         
25764 -1         -1           ⋯ -1              -1            -1         
      es_damage ce_damage ice_damage mguk_damage tc_damage engine_blown
25759 -1        -1        -1         -1          -1        -1          
25760 -1        -1        -1         -1          -1        -1          
25761 -1        -1        -1         -1          -1        -1          
25762 -1        -1        -1         -1          -1        -1          
25763 -1        -1        -1         -1          -1        -1          
25764 -1        -1        -1         -1          -1        -1          
      engine_seized
25759 -1           
25760 -1           
25761 -1           
25762 -1           
25763 -1           
25764 -1

Observamos que lapNum tiene un valor mayor al esperado (4), ya que arranca a contabilizar las vueltas desde 0, por lo que el lapNum = 5 corresponderia a la vuelta 6. Además observamos que los valores de muchas variables toman el valor "-1", lo cual es inconsistente en algunas variables como "lap_distance", por lo que aunque debemos eliminar todos los registros de esta vuelta, vamos a realizar un breve análisis que nos permita comprender que sucedió.

In [7]:
wil <- wil %>%
  select(
    carId,
    lapNum,
    lap_time,
    lap_distance,
    lap_time_invalid,
    binIndex,
    velocity_X,
    velocity_Y,
    velocity_Z
  )

Para trabajar más cómodos, decidimos eliminar todas aquellas columnas que no vamos a utilizar ni ahora ni a futuro.

In [8]:
inconsistencia <- wil %>%

  select(where(is.numeric)) %>%

  summarise(
    across(
      everything(),
      ~ sum(. == -1, na.rm = TRUE)
    )
  )

inconsistencia

lapNum lap_time lap_distance lap_time_invalid binIndex velocity_X velocity_Y
1 0      3237     3237         3237             0        3237       3237      
  velocity_Z
1 3237

Creamos un pequeño dataset que nos permite ver cuantas de las variables númericas existentes en nuestro dataset tienen un valor igual a "-1". Al imprimirla observamos que 6 de las 8 variables tienen 3237 registros de este tipo. Sospechamos que cuando una de estas variables toma el valor -1, las otras 5 también.

In [9]:
sospechosas <- c(
  "lap_time",
  "lap_distance",
  "lap_time_invalid",
  "velocity_X",
  "velocity_Y",
  "velocity_Z"
)


conteo_sospechosas <- wil %>%
  summarise(
    cant_sospechosas = sum(if_all(all_of(sospechosas), ~ . == -1), na.rm = TRUE)
  ) %>%
  pull()

conteo_sospechosas

[1] 3237

Se confirma nuestra teoría. Además, por lo leído en el archivo de especificaciones de EA SPORTS F1, este valor suele significar que no se ha cruzado la línea de meta, lo que puede indicar que estos valores pertenezcan a:

1. Una vuelta de calentamiento.
2. Los metros que sigue recorriendo el automóvil antes de frenar, luego de cruzar la meta.

In [10]:
final_valido <- wil %>%
  filter(lapNum == 5) %>%
  group_by(lap_time_invalid) %>%
  summarise(
    N_Registros = n(),
    .groups = 'drop'
  )

  final_valido

lap_time_invalid N_Registros
1 -1               3237       
2  0               1057

Confirmamos que todos estos valores pertenecen a la vuelta 6 (lapNum = 5), lo que indicaria que todos estos registros se deben a la razón 2.

In [11]:
wil <- wil %>%
  filter(lapNum < 5)

Borramos esta vuelta.

In [12]:
mcl <- read.csv("mcclaren.csv")

In [13]:
head(mcl)

carId  trackId    trackLength lapIndex lapNum lapFlag binIndex validBin
1 Alpine Interlagos 4294        0        21     0       0        1       
2 Alpine Interlagos 4294        0        21     0       1        1       
3 Alpine Interlagos 4294        0        21     0       2        1       
4 Alpine Interlagos 4294        0        21     0       3        1       
5 Alpine Interlagos 4294        0        21     0       4        1       
6 Alpine Interlagos 4294        0        21     0       5        1       
  lap_number lap_distance ⋯ gear_box_damage engine_damage mguh_damage es_damage
1 21         0.40625      ⋯ 0               0             0           0        
2 21         1.41602      ⋯ 0               0             0           0        
3 21         2.42578      ⋯ 0               0             0           0        
4 21         3.43555      ⋯ 0               0             0           0        
5 21         4.44531      ⋯ 0               0             0           0        
6 21         5.45703      ⋯ 0               0             0           0        
  ce_damage ice_damage mguk_damage tc_damage engine_blown engine_seized
1 0         0          0           0         0            0            
2 0         0          0           0         0            0            
3 0         0          0           0         0            0            
4 0         0          0           0         0            0            
5 0         0          0           0         0            0            
6 0         0          0           0         0            0

In [14]:
mcl$carId <- "MCL"

In [15]:
mcl$lapNum <- mcl$lapNum - 21

In [16]:
tail(mcl)

carId trackId    trackLength lapIndex lapNum lapFlag binIndex validBin
25759 MCL   Interlagos 4294        5        5      0       4288     0       
25760 MCL   Interlagos 4294        5        5      0       4289     0       
25761 MCL   Interlagos 4294        5        5      0       4290     0       
25762 MCL   Interlagos 4294        5        5      0       4291     0       
25763 MCL   Interlagos 4294        5        5      0       4292     0       
25764 MCL   Interlagos 4294        5        5      0       4293     0       
      lap_number lap_distance ⋯ gear_box_damage engine_damage mguh_damage
25759 -1         -1           ⋯ -1              -1            -1         
25760 -1         -1           ⋯ -1              -1            -1         
25761 -1         -1           ⋯ -1              -1            -1         
25762 -1         -1           ⋯ -1              -1            -1         
25763 -1         -1           ⋯ -1              -1            -1         
25764 -1         -1           ⋯ -1              -1            -1         
      es_damage ce_damage ice_damage mguk_damage tc_damage engine_blown
25759 -1        -1        -1         -1          -1        -1          
25760 -1        -1        -1         -1          -1        -1          
25761 -1        -1        -1         -1          -1        -1          
25762 -1        -1        -1         -1          -1        -1          
25763 -1        -1        -1         -1          -1        -1          
25764 -1        -1        -1         -1          -1        -1          
      engine_seized
25759 -1           
25760 -1           
25761 -1           
25762 -1           
25763 -1           
25764 -1

In [17]:
mcl <- mcl %>%
  filter(lapNum < 5)

In [18]:
mcl <- mcl %>%
  select(
    carId,
    lapNum,
    lap_time,
    lap_distance,
    lap_time_invalid,
    binIndex,
    velocity_X,
    velocity_Y,
    velocity_Z
  )

In [19]:
red <- read.csv("redbull.csv")

In [20]:
head(red)

carId  trackId    trackLength lapIndex lapNum lapFlag binIndex validBin
1 Alpine Interlagos 4294        0        16     0       0        1       
2 Alpine Interlagos 4294        0        16     0       1        1       
3 Alpine Interlagos 4294        0        16     0       2        1       
4 Alpine Interlagos 4294        0        16     0       3        1       
5 Alpine Interlagos 4294        0        16     0       4        1       
6 Alpine Interlagos 4294        0        16     0       5        1       
  lap_number lap_distance ⋯ gear_box_damage engine_damage mguh_damage es_damage
1 16         2.81250      ⋯ 0               0             0           0        
2 16         2.81250      ⋯ 0               0             0           0        
3 16         2.81250      ⋯ 0               0             0           0        
4 16         3.82617      ⋯ 0               0             0           0        
5 16         4.83984      ⋯ 0               0             0           0        
6 16         5.85352      ⋯ 0               0             0           0        
  ce_damage ice_damage mguk_damage tc_damage engine_blown engine_seized
1 0         0          0           0         0            0            
2 0         0          0           0         0            0            
3 0         0          0           0         0            0            
4 0         0          0           0         0            0            
5 0         0          0           0         0            0            
6 0         0          0           0         0            0

In [21]:
red$carId <-"RED"

In [22]:
red$lapNum <- red$lapNum - 16

In [23]:
tail(red)

carId trackId    trackLength lapIndex lapNum lapFlag binIndex validBin
25759 RED   Interlagos 4294        5        5      0       4288     0       
25760 RED   Interlagos 4294        5        5      0       4289     0       
25761 RED   Interlagos 4294        5        5      0       4290     0       
25762 RED   Interlagos 4294        5        5      0       4291     0       
25763 RED   Interlagos 4294        5        5      0       4292     0       
25764 RED   Interlagos 4294        5        5      0       4293     0       
      lap_number lap_distance ⋯ gear_box_damage engine_damage mguh_damage
25759 -1         -1           ⋯ -1              -1            -1         
25760 -1         -1           ⋯ -1              -1            -1         
25761 -1         -1           ⋯ -1              -1            -1         
25762 -1         -1           ⋯ -1              -1            -1         
25763 -1         -1           ⋯ -1              -1            -1         
25764 -1         -1           ⋯ -1              -1            -1         
      es_damage ce_damage ice_damage mguk_damage tc_damage engine_blown
25759 -1        -1        -1         -1          -1        -1          
25760 -1        -1        -1         -1          -1        -1          
25761 -1        -1        -1         -1          -1        -1          
25762 -1        -1        -1         -1          -1        -1          
25763 -1        -1        -1         -1          -1        -1          
25764 -1        -1        -1         -1          -1        -1          
      engine_seized
25759 -1           
25760 -1           
25761 -1           
25762 -1           
25763 -1           
25764 -1

In [24]:
red <- red %>%
  filter(lapNum < 5)

In [25]:
red <- red %>%
  select(
    carId,
    lapNum,
    lap_time,
    lap_distance,
    lap_time_invalid,
    binIndex,
    velocity_X,
    velocity_Y,
    velocity_Z
  )

In [26]:
alp <- read.csv("alpine.csv")

In [27]:
head(alp)

carId  trackId    trackLength lapIndex lapNum lapFlag binIndex validBin
1 Alpine Interlagos 4294        0        10     0       0        1       
2 Alpine Interlagos 4294        0        10     0       1        1       
3 Alpine Interlagos 4294        0        10     0       2        1       
4 Alpine Interlagos 4294        0        10     0       3        1       
5 Alpine Interlagos 4294        0        10     0       4        1       
6 Alpine Interlagos 4294        0        10     0       5        1       
  lap_number lap_distance ⋯ gear_box_damage engine_damage mguh_damage es_damage
1 10         1.63281      ⋯ 0               0             0           0        
2 10         1.63281      ⋯ 0               0             0           0        
3 10         2.62500      ⋯ 0               0             0           0        
4 10         3.61719      ⋯ 0               0             0           0        
5 10         4.60938      ⋯ 0               0             0           0        
6 10         5.60156      ⋯ 0               0             0           0        
  ce_damage ice_damage mguk_damage tc_damage engine_blown engine_seized
1 0         0          0           0         0            0            
2 0         0          0           0         0            0            
3 0         0          0           0         0            0            
4 0         0          0           0         0            0            
5 0         0          0           0         0            0            
6 0         0          0           0         0            0

In [28]:
alp$carId <- "ALP"

In [29]:
alp$lapNum <- alp$lapNum - 10

In [30]:
tail(alp)

carId trackId    trackLength lapIndex lapNum lapFlag binIndex validBin
30053 ALP   Interlagos 4294        6        6      0       4288     0       
30054 ALP   Interlagos 4294        6        6      0       4289     0       
30055 ALP   Interlagos 4294        6        6      0       4290     0       
30056 ALP   Interlagos 4294        6        6      0       4291     0       
30057 ALP   Interlagos 4294        6        6      0       4292     0       
30058 ALP   Interlagos 4294        6        6      0       4293     0       
      lap_number lap_distance ⋯ gear_box_damage engine_damage mguh_damage
30053 -1         -1           ⋯ -1              -1            -1         
30054 -1         -1           ⋯ -1              -1            -1         
30055 -1         -1           ⋯ -1              -1            -1         
30056 -1         -1           ⋯ -1              -1            -1         
30057 -1         -1           ⋯ -1              -1            -1         
30058 -1         -1           ⋯ -1              -1            -1         
      es_damage ce_damage ice_damage mguk_damage tc_damage engine_blown
30053 -1        -1        -1         -1          -1        -1          
30054 -1        -1        -1         -1          -1        -1          
30055 -1        -1        -1         -1          -1        -1          
30056 -1        -1        -1         -1          -1        -1          
30057 -1        -1        -1         -1          -1        -1          
30058 -1        -1        -1         -1          -1        -1          
      engine_seized
30053 -1           
30054 -1           
30055 -1           
30056 -1           
30057 -1           
30058 -1

In [31]:
alp <- alp %>%
  filter(lapNum < 5)

In [32]:
alp <- alp %>%
  select(
    carId,
    lapNum,
    lap_time,
    lap_distance,
    lap_time_invalid,
    binIndex,
    velocity_X,
    velocity_Y,
    velocity_Z
  )

In [33]:
mer <- read.csv("mercedes.csv")

In [34]:
head(mer)

carId  trackId    trackLength lapIndex lapNum lapFlag binIndex validBin
1 Alpine Interlagos 4294        0        5      0       0        1       
2 Alpine Interlagos 4294        0        5      0       1        1       
3 Alpine Interlagos 4294        0        5      0       2        1       
4 Alpine Interlagos 4294        0        5      0       3        1       
5 Alpine Interlagos 4294        0        5      0       4        1       
6 Alpine Interlagos 4294        0        5      0       5        1       
  lap_number lap_distance ⋯ gear_box_damage engine_damage mguh_damage es_damage
1 5          3.34570      ⋯ 0               0             0           0        
2 5          3.34570      ⋯ 0               0             0           0        
3 5          3.34570      ⋯ 0               0             0           0        
4 5          3.34570      ⋯ 0               0             0           0        
5 5          4.41357      ⋯ 0               0             0           0        
6 5          5.48145      ⋯ 0               0             0           0        
  ce_damage ice_damage mguk_damage tc_damage engine_blown engine_seized
1 0         0          0           0         0            0            
2 0         0          0           0         0            0            
3 0         0          0           0         0            0            
4 0         0          0           0         0            0            
5 0         0          0           0         0            0            
6 0         0          0           0         0            0

In [35]:
mer$carId <-"MER"

In [36]:
mer$lapNum <- mer$lapNum - 5

In [37]:
tail(mer)

carId trackId    trackLength lapIndex lapNum lapFlag binIndex validBin
25759 MER   Interlagos 4294        5        5      0       4288     0       
25760 MER   Interlagos 4294        5        5      0       4289     0       
25761 MER   Interlagos 4294        5        5      0       4290     0       
25762 MER   Interlagos 4294        5        5      0       4291     0       
25763 MER   Interlagos 4294        5        5      0       4292     0       
25764 MER   Interlagos 4294        5        5      0       4293     0       
      lap_number lap_distance ⋯ gear_box_damage engine_damage mguh_damage
25759 -1         -1           ⋯ -1              -1            -1         
25760 -1         -1           ⋯ -1              -1            -1         
25761 -1         -1           ⋯ -1              -1            -1         
25762 -1         -1           ⋯ -1              -1            -1         
25763 -1         -1           ⋯ -1              -1            -1         
25764 -1         -1           ⋯ -1              -1            -1         
      es_damage ce_damage ice_damage mguk_damage tc_damage engine_blown
25759 -1        -1        -1         -1          -1        -1          
25760 -1        -1        -1         -1          -1        -1          
25761 -1        -1        -1         -1          -1        -1          
25762 -1        -1        -1         -1          -1        -1          
25763 -1        -1        -1         -1          -1        -1          
25764 -1        -1        -1         -1          -1        -1          
      engine_seized
25759 -1           
25760 -1           
25761 -1           
25762 -1           
25763 -1           
25764 -1

In [38]:
mer <- mer %>%
  filter(lapNum < 5)

In [39]:
mer <- mer %>%
  select(
    carId,
    lapNum,
    lap_time,
    lap_distance,
    lap_time_invalid,
    binIndex,
    velocity_X,
    velocity_Y,
    velocity_Z
  )

In [40]:
mer <- mer %>%
  mutate(
    sector = case_when(
      binIndex >= 0 & binIndex <= 1430 ~ "S1",
      binIndex >= 1431 & binIndex <= 2861 ~ "S2",
      binIndex >= 2862 & binIndex <= 4293 ~ "S3",
    ))

In [41]:
fer <- read.csv("ferrari.csv")

In [42]:
head(fer)

carId  trackId    trackLength lapIndex lapNum lapFlag binIndex validBin
1 Alpine Interlagos 4294        0        31     0       0        1       
2 Alpine Interlagos 4294        0        31     0       1        1       
3 Alpine Interlagos 4294        0        31     0       2        1       
4 Alpine Interlagos 4294        0        31     0       3        1       
5 Alpine Interlagos 4294        0        31     0       4        1       
6 Alpine Interlagos 4294        0        31     0       5        1       
  lap_number lap_distance ⋯ gear_box_damage engine_damage mguh_damage es_damage
1 31         0.93750      ⋯ 0               0             0           0        
2 31         1.80313      ⋯ 0               0             0           0        
3 31         2.66875      ⋯ 0               0             0           0        
4 31         3.53438      ⋯ 0               0             0           0        
5 31         4.40000      ⋯ 0               0             0           0        
6 31         5.26562      ⋯ 0               0             0           0        
  ce_damage ice_damage mguk_damage tc_damage engine_blown engine_seized
1 0         0          0           0         0            0            
2 0         0          0           0         0            0            
3 0         0          0           0         0            0            
4 0         0          0           0         0            0            
5 0         0          0           0         0            0            
6 0         0          0           0         0            0

In [43]:
fer$carId <- "FER"

In [44]:
fer$lapNum <- fer$lapNum - 31

In [45]:
tail(fer)

carId trackId    trackLength lapIndex lapNum lapFlag binIndex validBin
21465 FER   Interlagos 4294        4        4      0       4288     0       
21466 FER   Interlagos 4294        4        4      0       4289     0       
21467 FER   Interlagos 4294        4        4      0       4290     0       
21468 FER   Interlagos 4294        4        4      0       4291     0       
21469 FER   Interlagos 4294        4        4      0       4292     0       
21470 FER   Interlagos 4294        4        4      0       4293     0       
      lap_number lap_distance ⋯ gear_box_damage engine_damage mguh_damage
21465 -1         -1           ⋯ -1              -1            -1         
21466 -1         -1           ⋯ -1              -1            -1         
21467 -1         -1           ⋯ -1              -1            -1         
21468 -1         -1           ⋯ -1              -1            -1         
21469 -1         -1           ⋯ -1              -1            -1         
21470 -1         -1           ⋯ -1              -1            -1         
      es_damage ce_damage ice_damage mguk_damage tc_damage engine_blown
21465 -1        -1        -1         -1          -1        -1          
21466 -1        -1        -1         -1          -1        -1          
21467 -1        -1        -1         -1          -1        -1          
21468 -1        -1        -1         -1          -1        -1          
21469 -1        -1        -1         -1          -1        -1          
21470 -1        -1        -1         -1          -1        -1          
      engine_seized
21465 -1           
21466 -1           
21467 -1           
21468 -1           
21469 -1           
21470 -1

En este caso si tenemos la cantidad correcta de vueltas, pero como podemos ver hay muchisimos valores "-1", por lo que debemos borrar esta vuelta y generarla de manera artificial.

In [46]:
fer <- fer %>%
  filter(lapNum < 4)

In [47]:
fer <- fer %>%
  select(
    carId,
    lapNum,
    lap_time,
    lap_distance,
    lap_time_invalid,
    binIndex,
    velocity_X,
    velocity_Y,
    velocity_Z
  )

In [48]:
vueltas_anuladas_fer <- fer %>%
  group_by(lapNum) %>%
  summarise(lap_time_invalid = max(lap_time_invalid))

Para generar de forma artificial la quinta vuelta, debemos realizarlo promediando los valores de las vueltas validas, por lo que visualizamos cuales son (las que tienen lap_time_valid = 0)

In [49]:
vueltas_anuladas_fer

lapNum lap_time_invalid
1 0      0               
2 1      1               
3 2      1               
4 3      0

Solamente podemos trabajar con la vuelta 1 y 4.

In [50]:
vueltas_validas <- fer %>%
  filter(lapNum %in% c(0, 3))

In [51]:
vuelta_5 <- vueltas_validas %>%
  group_by(binIndex) %>%
  summarise(
    lap_distance = mean(lap_distance, na.rm = TRUE),
    lap_time = mean(lap_time, na.rm = TRUE),
    velocity_X = mean(velocity_X, na.rm = TRUE),
    velocity_Y = mean(velocity_Y, na.rm = TRUE),
    velocity_Z = mean(velocity_Z, na.rm = TRUE),

    carId = "FER",
    lap_time_invalid = 0
  ) %>%

  ungroup()%>%

  mutate(
    lapNum = 4
  )

Las filtramos en el dataset "vueltas_validas" y construimos la "vuelta_5" calculando los promedios en la variables indicadas, despues le pasamos el valor del id, que la vuelta es valida y que es la número 5.

In [52]:
fer <- bind_rows(fer, vuelta_5)

La agregamos a nuestro dataset de ferrari.

In [53]:
telemetria <- bind_rows(fer, red, mcl, mer, alp, wil)

Únimos los 6 datasets en uno único.

In [54]:
telemetria <- telemetria %>%
  mutate(
    sector = case_when(
      binIndex >= 0 & binIndex <= 1430 ~ "S1",
      binIndex >= 1431 & binIndex <= 2861 ~ "S2",
      binIndex >= 2862 & binIndex <= 4293 ~ "S3",
    ),
    track_lenght = 4294
    )

Creamos la columna sector, indicando el valor que toma según el binIndex.

In [55]:
telemetria <- telemetria %>%
  mutate(
    velocity_total_ms = sqrt(
      velocity_X^2 +
      velocity_Y^2 +
      velocity_Z^2
    ),

    velocidad = velocity_total_ms * 3.6
  )

telemetria <- telemetria %>%
  select(-velocity_total_ms)

In [56]:
inconsistencia_global <- telemetria %>%

  select(where(is.numeric)) %>%

  summarise(
    across(
      everything(),
      ~ sum(. == -1, na.rm = TRUE)
    )
  )

inconsistencia_global

lapNum lap_time lap_distance lap_time_invalid binIndex velocity_X velocity_Y
1 0      0        0            0                0        0          0         
  velocity_Z track_lenght velocidad
1 0          0            0

In [57]:
telemetria <- telemetria %>%
  mutate(lap_number = lapNum + 1)

In [60]:
registros <- nrow(telemetria)
print(paste("cant. registros:", registros))

escuderias <- unique(telemetria$carId)
print(escuderias)

cant_lap <- unique(telemetria$lapNum)
print(cant_lap)

cant_lap <- unique(telemetria$lap_number)
print(cant_lap)

any(is.na(telemetria))

[1] "cant. registros: 128820"
[1] "FER" "RED" "MCL" "MER" "ALP" "WIL"
[1] 0 1 2 3 4
[1] 1 2 3 4 5


[1] FALSE

In [59]:
options(repr.plot.width = 20, repr.plot.height = 14)